Docker.md

```diff
@@ -2,8 +2,19 @@
 
 ### Publishing ports
 
-Ports for services should be explicitly published to the loopback interface, 127.0.0.1 to prevent docker from binding them to 0.0.0.0.
-For example, when using postgres or elasticsearch, we might want to make them available to developers, but in the case that we deploy on a public server for testing we don't want them to be exposed.
+By default, containers within the same docker-compose network are able to connect to services running
+on other containers on any port by using the container name, as long as the container is listening
+on 0:0:0:0.
+
+The [`EXPOSE` directive isn't needed](https://docs.docker.com/engine/reference/builder/#expose), although
+some tools ([such as the nginx proxy companion](https://github.com/nginx-proxy/acme-companion#step-3---proxied-containers))
+use it to define a contract between different containers. Use it if you think it will be necessary.
```

Comment:
Right, `EXPOSE` is generally not needed in the Dockerfile, but `expose` can be useful in Docker Compose files.

Reply:
interesting. what does expose in docker-compose configuration do here? Is it the same as `docker run --expose` and `EXPOSE` in the Dockerfile, but just that you can apply it dynamically at runtime rather than hard-coding into the Dockerfile?

Reply:
Yes

Reply:
It isn’t useful but for documentation and tools such as `nginx-proxy`. Distinguish between `EXPOSE` builder directive and `expose` compose option in https://github.com/metabrainz/guidelines/pull/18/commits/627f33d81e431191660b62ef2d35102f35508841.